### ISSUE TITLE
Building with charm version 3.x does not install on bionic (Python >=3.7 is required, 3.6.9 is present)

### ISSUE THREAD

**Issue by esunar** [canonical/charmcraft#888]
When we pack with charm 3.x does not install on bionic

[Running functional tests with 3.x:](https://pastebin.ubuntu.com/p/ZyKf6V2hFg/)
```
type: charm
parts:
  charm:
    source: src
    plugin: reactive
    build-snaps: 
      - charm
bases:
    - build-on:
        - name: ubuntu
          channel: "22.04"
          architectures: ["amd64"]
      run-on:
        - name: ubuntu
          channel: "22.04"
          architectures:
              - amd64
        - name: ubuntu
          channel: "20.04"
          architectures:
              - amd64
        - name: ubuntu
          channel: "18.04"
          architectures:
              - amd64

```
[Running functional tests with 2.x:](https://pastebin.ubuntu.com/p/6nJVMzJZDN/)

```
type: charm
parts:
  charm:
    source: src
    plugin: reactive
    build-snaps: 
      - charm/2.x/stable
bases:
    - build-on:
        - name: ubuntu
          channel: "22.04"
          architectures: ["amd64"]
      run-on:
        - name: ubuntu
          channel: "22.04"
          architectures:
              - amd64
        - name: ubuntu
          channel: "20.04"
          architectures:
              - amd64
        - name: ubuntu
          channel: "18.04"
          architectures:
              - amd64

```

created a demo branch for easy tesing:

```
git clone -b charmcraft-bionic-bug https://git.launchpad.net/~esunar/charm-graylog
make charm3 #fails
make charm2 #sucess
```

**Comment by sergiusens** [comment]
The error is in your logs: 
    
    2022-10-05 18:57:46 [ERROR] unit-graylog-0.log: 2022-10-05 15:57:45 WARNING unit.graylog/0.install logger.go:60 pip requires Python '>=3.7' but the running Python is 3.6.9

The root of this is that your are building in the future (Ubuntu 22.04) and targeting the past (Ubuntu 20.04 and Ubuntu 18.04). This generally can work, but in most cases won't as the Python interpreter is provided by the OS.

**Comment by sergiusens** [comment]
charm tools changed last week which is probably what broke this. Have you logged an issue against charm tools?

**Comment by fnordahl** [comment]
The build on bionic / 18.04 failure was fixed in juju/charm-tools#641 and adds further ammunition to why this should be fixed in charmcraft as reported in #878.

The recent move of charm-tools from 2.x to 3.x was forced by the need to support charms running on newer releases such as focal/jammy/kinetic. Before the change, the whole purpose of charmcraft bases was circumvented by the fact that charm-tools 2.x would use Python from a Core18 base to build a charms wheelhouse.

This is of course not maintainable, and did also prevent building functioning charms for recent releases.

**Comment by facundobatista** [comment]
Hello! Thanks for the report. 

So, both the first issue and the second issue are addressed (the former has a wrong config, the later is fixed in `charm` tool). We need to improve the overall situation, and that's tracked in #878. 

Nothing is left for this issue, so closing it. Thanks again!
